src/service/xhr-impl.js

```diff
@@ -94,6 +96,15 @@ export class Xhr {
   constructor(win) {
     /** @const {!Window} */
     this.win = win;
+
+    // TODO(@jridgewell, #9048): Once we figure out if ampdoc can always be
+    // found, remove the try catch.
+    try {
+      this.ampdoc_ = ampdocServiceFor(win).getAmpDoc();
+    } catch (e) {
+      this.ampdoc_ = null;
+      dev().error('XHR', e, 'XHR service failed to find ampdoc');
```

Comment:
That's what I was thinking. Though I don't agree that it's only needed in single-doc mode. Is there a way to get the parent amp-doc while in PWA?